src/Akkatecture/Aggregates/CommandResults/CommandResult.cs

```diff
@@ -0,0 +1,56 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2015-2021 Rasmus Mikkelsen
+// Copyright (c) 2015-2021 eBay Software Foundation
+//     Modified from original source https://github.com/eventflow/EventFlow
+// Copyright (c) 2018 - 2021 Lutando Ngqakaza
+//     https://github.com/Lutando/Akkatecture 
+// Copyright (c) 2022 AfterLutz Contributors
+//     https://github.com/AfterLutz/Akketecture
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Collections.Generic;
+using Akkatecture.Commands;
+
+namespace Akkatecture.Aggregates.CommandResults;
+
+public class CommandResult : ICommandResult
+{
+    public static ICommandResult SucceedWith(ICommand command) =>
```

Comment:
It seems a little odd to me to reference derived classes in a base class.  I wonder if it would be cleaner to create extension methods on the ICommand interface instead of keeping these here.

Reply:
We're using the base class as a factory for the derived class. It might make more sense if the base class were abstract, but then that would reduce the flexibility of it too.